apps/communications/dialer/test/marionette/lib/views/activities/views.js

```diff
@@ -10,15 +10,24 @@ function ActivitiesView(client) {
 
 ActivitiesView.prototype = {
   pickCreateNew: function() {
-      this.client.waitFor(function() {
-        return this.accessors.addNewContactMenuItem.enabled();
-      }.bind(this));
-      var button = this.accessors.addNewContactMenuItem;
-      button.tap();
-
+      this._openElement(this.accessors.addNewContactMenuItem);
+      
       var contacts = this.client.loader.getAppClass('contacts');
       contacts.switchToCreateNewContactActivity();
       return contacts;
+  },
+
+  pickAddToContact: function(){
+      this._openElement(this.accessors.addToContactMenuItem);
+
+      var contacts = this.client.loader.getAppClass('contacts');
+      contacts.switchToApp();
+      return contacts;
+  },
+
+  _openElement: function(accessorToTap){
+      var button = accessorToTap;
+      button.tap();
```

Comment:
Hi Johan, sorry for asking so much, but is there a way I can compare two accessors? (sorry if this is a silly question :S)I wanted to do something like:
accessorToTap.tap();
      var buttonTapped = accessorToTap.getAttribute('data-l10n-id');
      var contacts = this.client.loader.getAppClass('contacts');
     
      switch (buttonTapped) {
        case 'createNewContact':
          contacts.switchToCreateNewContactActivity();
          break;
        case 'addToExistingContact':
          contacts.switchToApp();
          break;
      }
      return contacts;  
  }

but comparing the accessorToTap with the corresponding accessor (this.accessors.addToContactMenuItem) so I can go to the correct part of contacts app, but did find the way...And I guess that this solution I did so far is not good since it relays on the text of the buttons that can change...